### ISSUE TITLE
Support for AWS Cloudwatch Logs Exporter

### ISSUE THREAD

**Issue by Imshelledin21** [grafana/alloy#627]
### Request

Bring the [AWS Cloudwatch Logs Exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/awscloudwatchlogsexporter) component from opentelemetry-collector-contrib into Alloy

### Use case

Ability to send scraped logs to AWS Cloudwatch

**Comment by hainenber** [comment]
I wonder if [`prometheus.exporter.cloudwatch`](https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.exporter.cloudwatch/) is sufficient for your usecase 👀 ?

**Comment by Imshelledin21** [comment]
I'm not wanting to scrape metrics/logs from Cloudwatch, but rather send scraped logs to Cloudwatch.

**Comment by irizzant** [comment]
This would be very nice indeed

**Comment by hhappel** [comment]
Seems to be related / likely a duplicate #1080 ?

**Comment by adamcstephens** [comment]
> Seems to be related / likely a duplicate [#1080](https://github.com/grafana/alloy/issues/1080) ?

That appears to be for receiving _from_ cloudwatch, while this should for sending _to_ cloudwatch.

**Comment by franklouwers** [comment]
Any plans to support this (or workarounds)

**Comment by jsmcnair** [comment]
I've been looking into this and, although I've not got a PoC together yet I think it's possible.

AWS supports open telemetry endpoints for CloudWatch: https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-OTLPEndpoint.html. 

There are some considerations:
* Sigv4Authentication extension - https://grafana.com/docs/alloy/latest/reference/components/otelcol/otelcol.auth.sigv4/
* HTTP rather than gRPC - https://grafana.com/docs/alloy/latest/reference/components/otelcol/otelcol.exporter.otlphttp/
* Gzip Compression (over 1MB) - This appears to be enabled by default
* `x-aws-log-group` and `x-aws-log-stream` HTTP headers: https://grafana.com/docs/alloy/latest/reference/components/otelcol/otelcol.exporter.otlphttp/#client

**Comment by ryepup** [comment]
@jsmcnair I did a similar PoC recently, and was able to get logs flowing.

My PoC worked, but could probably be simplified. Here's what I did:

1. made an EC2 instance and installed nginx and alloy
2. made a cloudwatch log group `NGINXLogGroup` and stream `NGINXLogStream` in `us-west-2`
3. made a new IAM role `nginx_log_writer` that allows `logs:PutLogEvents` for my stream (`arn:aws:logs:us-west-2:REDACTED:log-group:NGINXLogGroup:log-stream:NGINXLogs`)
4. added more policy config so my EC2 could `sts:AssumeRole` that `nginx_log_writer` role. I was testing [cross-account policies](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_evaluation-logic-cross-account.html), which needed more IAM pieces.
5. setup alloy with this pipeline: `loki.source.file` -> `otelcol.receiver.loki` -> `otelcol.exporter.otlphttp` + `otelcol.auth.sigv4`

Here's how my config ended up:

```alloy
loki.source.file "nginx_logs" {
  targets = [
    {__path__ = "/var/log/nginx/access.log"},
    {__path__ = "/var/log/nginx/error.log"},
  ]
  forward_to = [otelcol.receiver.loki.nginx_logs.receiver]
}

otelcol.receiver.loki "nginx_logs" {
  output {
    logs = [otelcol.exporter.otlphttp.cloudwatch.input]
  }
}

otelcol.exporter.otlphttp "cloudwatch" {
  client {
    endpoint = "https://logs.us-west-2.amazonaws.com" // <-- REGION MUST MATCH CLOUDWATCH
    auth     = otelcol.auth.sigv4.default.handler
    headers = {
      "x-aws-log-group" = "NGINXLogGroup", // <-- MUST MATCH CLOUDWATCH
      "x-aws-log-stream" = "NGINXLogStream", // <--MUST MATCH CLOUDWATCH
    }
  }
}

otelcol.auth.sigv4 "default" {
  service = "logs"

  assume_role {
    arn = "arn:aws:iam::REDACTED:role/nginx_log_writer" // <-- MUST MATCH IAM ROLE
    sts_region = "us-west-2" // <-- REGION MUST MATCH CLOUDWATCH
  }
}
```

That gives me JSON logs in my cloudwatch stream, and I can enrich those by adding a `otelcol.processor.transform` step in between `otelcol.receiver.loki` and `otelcol.exporter.otlphttp`

I'm guessing there are easier ways to navigate IAM to eliminate the need for `assume_role`, and the `otelcol.receiver.loki` step might be optional; I want it to add attributes, but your use case might not need it.

**Comment by jsmcnair** [comment]
Yes - I tried this and it worked for me too, including using cross-account access and IRSA. Thanks for writing it up!
